figures/scripts/transcriptomic-overview.R

```diff
@@ -0,0 +1,315 @@
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# Declare output directory
+output_dir <- file.path(root_dir, "figures", "pngs")
+
+# Final figure filename
+output_png <- file.path(output_dir, "transcriptomic-overview.png")
+
+# Data directory
+data_dir <- file.path(root_dir, "data")
+
+# Analysis directory
+analyses_dir <- file.path(root_dir, "analyses")
+
+# Read in clinical data
+histologies_df <- read_tsv(file.path(data_dir, "pbta-histologies.tsv"),
+                           guess_max = 10000)
+
+#### Temporary PNG names -------------------------------------------------------
+# Because we are mixing Heatmaps and ggplots in this figure, one of the best
+# ways to control sizes is to create PNGs and then use multipanelfigure to
+# put the final figure together.
+#
+# Here we'll set up the file names used throughout
+umap_png <- file.path(output_dir, "temp_umap.png")
+gsva_png <- file.path(output_dir, "temp_gsva.png")
+deconv_png <- file.path(output_dir, "temp_deconv.png")
+legend_png <- file.path(output_dir, "temp_legend.png")
+
+#### Color palettes ------------------------------------------------------------
+
+palette_dir <- file.path(root_dir, "figures", "palettes")
+histology_palette <- read_tsv(file.path(palette_dir,
+                                        "histology_color_palette.tsv"))
+divergent_palette <- read_tsv(file.path(palette_dir,
+                                        "divergent_color_palette.tsv"))
+gradient_palette <- read_tsv(file.path(palette_dir,
+                                       "gradient_color_palette.tsv"))
+
+# The NA color is consistent across palettes
+na_color <- divergent_palette %>% filter(color_names == "na_color")
+
+# The palettes we will use for heatmaps need NA color removed and specified
+# separately
+divergent_palette  <- divergent_palette %>% filter(color_names != "na_color")
+gradient_palette  <- gradient_palette %>% filter(color_names != "na_color")
+
+#### Column heatmap annotation -------------------------------------------------
+# Annotation bar for short histology - this is used in the GSVA and immune
+# deconvolution heatmaps
+
+# First get a data frame for the annotation and order samples by short histology
+short_histology_df <- histologies_df %>%
+  filter(experimental_strategy == "RNA-Seq",
+         RNA_library == "stranded") %>%
+  select(Kids_First_Biospecimen_ID,
+         "short histology" = short_histology) %>%
+  arrange(`short histology`) %>%
+  tibble::column_to_rownames("Kids_First_Biospecimen_ID") %>%
+  as.data.frame()
+
+# The colors supplied to HeatmapAnnotation need to be a named
+# list of named vectors
+annotation_colors <-  tibble::deframe(histology_palette)
+
+# Annotation bar intended for the top of the heatmap
+column_heatmap_annotation <- HeatmapAnnotation(
+  df = short_histology_df,
+  name = "short histology",
+  col = list("short histology" = annotation_colors),
+  na_col = na_color$hex_codes,
+  annotation_name_side = "left",
```

Comment:
I do like the annotation left  change, but looking at the final output, this does not seem necessary for the Immune deconvolution plot, and adds a lot of unneeded white space to the final plot. If both plots are included, then I would think the best option would be to have the name only on the GSVA plot, and assume the reader can follow to the right with the same same colors and location.

Reply:
I both agree with you and am sad because I moved things around a bit so we only had to create the annotation once 😢 no big deal - will change.